### ISSUE TITLE
Still having trouble Resetting the game

### ISSUE THREAD

**Issue by Sphilippe7** [ga-wdi-boston/game-project#386]
So I've been working on this all night but I feel as though I am getting nowhere. The last thing I tried was an example code I got from Google. I'll post my code below.

My game logic file:

```js
'use strict';

const api = require('../auth/api');

let currentPlayer;
let count = 1;


//Game Board Positioning
let gameBoardArray = [["box", "box", "box"],["box", "box", "box"],["box", "box", "box"]];

const disableGame = function() {
  $(".tl").off('click');
  $(".tm").off('click');
  $(".tr").off('click');
  $(".ml").off('click');
  $(".mm").off('click');
  $(".mr").off('click');
  $(".bl").off('click');
  $(".bm").off('click');
  $(".br").off('click');
};

const reset = function() {
  $("#reset").click(function () {
    $("#gameBoardArray").text("+");
	$("box").removeClass('disable');
	$("box").removeClass('o');
	$("box").removeClass('x');
	$("box").removeClass('btn-primary');
	$("box").removeClass('btn-info');
	count = 0;

  });
};

//Winning the game
let checkforwinner = function() {
  if((gameBoardArray[0][0] === "x" && gameBoardArray[0][1] === "x" && gameBoardArray[0][2] === "x")||
  (gameBoardArray[1][0] === "x" && gameBoardArray[1][1] === "x" && gameBoardArray[1][2] === "x") ||
  (gameBoardArray[2][0] === "x" && gameBoardArray[2][1] === "x" && gameBoardArray[2][2] === "x") ||
  (gameBoardArray[0][0] === "x" && gameBoardArray[1][0] === "x" && gameBoardArray[2][0] === "x") ||
  (gameBoardArray[0][1] === "x" && gameBoardArray[1][1] === "x" && gameBoardArray[2][1] === "x") ||
  (gameBoardArray[0][2] === "x" && gameBoardArray[1][2] === "x" && gameBoardArray[2][2] === "x") ||

  (gameBoardArray[0][0] === "x" && gameBoardArray[1][1] === "x" && gameBoardArray[2][2] === "x") ||

  (gameBoardArray[0][2] === "x" && gameBoardArray[1][1] === "x" && gameBoardArray[2][1] === "x")
  )
  {
    $('.display-winner').text('Player 1 Wins!');
    disableGame();
  }


if((gameBoardArray[0][0] === "o" && gameBoardArray[0][1] === "o" && gameBoardArray[0][2] === "o") ||
  (gameBoardArray[1][0] === "o" && gameBoardArray[1][1] === "o" && gameBoardArray[1][2] === "o") ||
  (gameBoardArray[2][0] === "o" && gameBoardArray[2][1] === "o" && gameBoardArray[2][2] === "o") ||
  (gameBoardArray[0][0] === "o" && gameBoardArray[1][0] === "o" && gameBoardArray[2][0] === "o") ||
  (gameBoardArray[0][1] === "o" && gameBoardArray[1][1] === "o" && gameBoardArray[2][1] === "o") ||
  (gameBoardArray[0][2] === "o" && gameBoardArray[1][2] === "o" && gameBoardArray[2][2] === "o") ||
  (gameBoardArray[0][0] === "o" && gameBoardArray[1][1] === "o" && gameBoardArray[2][2] === "o") ||
  (gameBoardArray[0][2] === "o" && gameBoardArray[1][1] === "o" && gameBoardArray[2][1] === "o")
  )
  {
  $('.display-winner').text('Player 2 Wins!');
  disableGame();
  }
//if 9 turns have passed and there is no winner then its a tie!
if(count === 9) {
  $('.display-winner').text('Its a Tie!');
  disableGame();
}
};
```js

**Comment by Sphilippe7** [comment]
Part of my API

`````` js
const resetgame = () =>
$.ajax({
  url: api.host + '/games',
  method: 'POST',
  headers: {
    Authorization: 'Token token=' + api.user.token,
  },
});

module.exports = {
  signUp,
  signIn,
  signOut,
  changePassword,
  creategame,
  updategame,
  resetgame
};
```js
``````

**Comment by Sphilippe7** [comment]
part of my events file:

`````` js
const resetgame = function (event) {
  event.preventDefault();
  api.resetgame()
    .then(ui.resetgameSuccess)
    .catch(ui.failure);
};


const addHandlers = () => {
  $('.sign-up-form').on('submit', onSignUp);
  $('.sign-in-form').on('submit', onSignIn);
  $('.sign-out-form').on('submit', onSignOut);
  $('.change-password-form').on('submit', onChangePassword);
  $('.creategame').on('click', creategame);
  $('.resetgame').on('reset', resetgame);
};

module.exports = {
  addHandlers,
};
```js
``````

**Comment by Sphilippe7** [comment]
This is how my game looks with the "Start a New Game" button
![screen shot 2016-11-17 at 8 01 24 pm](https://cloud.githubusercontent.com/assets/9563419/20414504/c0af67b0-ad00-11e6-88a9-cc9c7fbb9f88.png)

**Comment by raq929** [comment]
Cool. So, does any thing happen when you click the button? I recommend doing one thing at a time. Make sure the click handler is working by commenting out all the code in it and console logging 'click' or something. Once that works, add one line of code and make sure it does what you expect.
Right now, I can't see how your click handler for the reset button will be run. You'll have to add it to addHandlers, I think.

**Comment by gaand** [comment]
Do you have a form with the class `resetgame`?  Please post the html for whatever element has that class.

Also, does anything call the `reset` function from the game logic?  It seems to be intended to add a click handler.

Also, also, please post the code for `ui.resetgameSuccess`.

**Comment by Sphilippe7** [comment]
Sorry again for not responding to you both. I had problems with my charger so my Mac died @gaand  @raq929 . Ok @raq929 the button acts just as the create game button works when I click. And @gaand I do not have a form class for it, I just have it as a div on my html. I will post the code now

**Comment by Sphilippe7** [comment]
In my index.html file : 

``` js
<!--Creating a game-->

<button class="creategame" type="button">Create a Game!</button>


<!--To Reset a game-->

<button class="resetgame" type="button">Start a New Game!</button>
```

**Comment by Sphilippe7** [comment]
This is my whole ui.js file: 

``` js
'use strict';

const api = require('../config');
const logic = require('../game/logic');


const success = (data) => {
  $('#messages').text('success');
  console.log(data);
};


const signInSuccess = data => {
  api.user = data.user;
  success(data);
  console.log(data);
};

const signUpSuccess = data => {
  api.user = data.user;
  success(data);
  console.log(data);
};

const signOutSuccess = () => {
api.user = null;
  success(null);
  console.log('signOutSuccess');
};

const changePasswordSuccess = () => {
  success(null);
  console.log('changePasswordSuccess');
};

const failure = (error) => {
  $('#messages').text('fail');
  console.error(error);
};

const creategameSuccess = data => {
  api.game = data.game;
  success(data);
  logic.enableGame();
  console.log(data);
};

const resetgameSuccess = data => {
  api.game = data.game;
  success(data);
  logic.enableGame();
  console.log(data);
};

module.exports = {
  failure,
  success,
  signInSuccess,
  signUpSuccess,
  signOutSuccess,
  changePasswordSuccess,
  creategameSuccess,
  resetgameSuccess
};
```

**Comment by raq929** [comment]
So, does your create game button work the way you want it to?
I'm trying to figure out what the problem is. What do you expect to happen on reset game vs. what actually happens when you click.
Step 1 for me would be to comment out all the code in your click handler, and replace it with `console.log('click')`. If you click it and the console log works, then start adding things back in ONE thing at a time. It looks like you're trying to do a lot of things at once right now. Get one thing working, then move to the next.

**Comment by raq929** [comment]
So, think about what you want to happen when you click reset.
Is what you want happening? If no, how did you make that happen the first time around? Find the code that made it happen before and use it in your click handler.

**Comment by raq929** [comment]
How is this going?

**Comment by Sphilippe7** [comment]
Hey I tried a couple methods but it still does not work so I removed the reset button. I'm actively working on it now for another 5hrs so if you respond, I'll be here.

**Comment by raq929** [comment]
So what do you want the button to do? Shouldn't it behave almost exactly
like the create game button?

**Comment by Sphilippe7** [comment]
yes I just do not want it to be used till the game over. I either have it act as the create game button or it does not work at all. The point of the reset button is to be able to create a new game without reloading the page. As of now, that is the only way to play a new game @raq929

**Comment by Sphilippe7** [comment]
I recently tried some new code but it still does not work

**Comment by Sphilippe7** [comment]
My logic: 

``` js
const reset = function(){
    gameBoardArray = [["box", "box", "box"],["box", "box", "box"],["box", "box", "box"]];
    $('.box').off('click');
    $.each($('.box'), function(index, element) {
        $(element).html('');
        $(element).one('click', boxClick);
      } ) ;
      if (store.turn === "x"){
         store.turn = "o";
      } else {
         store.turn = "x";
      }
    $('.ttt-box').show();
    $('.gameEnd').text('');
};
```

**Comment by Sphilippe7** [comment]
and here is the button on my index.html file again:

``` js
<!--Reset Board-->
<button type="submit" class="reset">New Game</button>



<!--GameBoard-->
      <div class="ttt-box">
           <div class="row">
             <div class="col-xs-6 col-md-4 box tl"></div>
             <div class="col-xs-6 col-md-4 box tm"></div>
             <div class="col-xs-6 col-md-4 box tr"></div>
           </div>
           <div class="row">
             <div class="col-xs-6 col-md-4 box ml"></div>
             <div class="col-xs-6 col-md-4 box mm"></div>
             <div class="col-xs-6 col-md-4 box mr"></div>
           </div>
           <div class="row">
             <div class="col-xs-6 col-md-4 box bl"></div>
             <div class="col-xs-6 col-md-4 box bm"></div>
             <div class="col-xs-6 col-md-4 box br"></div>
           </div>
         </div>
```

**Comment by raq929** [comment]
What do you mean by 'it doesn't work?' what about it doesn't work? Are you
getting an error?
I strongly recommend following my previous advice. Comment out everything
in your click handler. Verify (using console.log or debugger) that your
click handler is working. Then add one line of code at a time, and verify
that the code does what you want using console.log or debugger.
That way, you can identify the problem. Right now, the handler is doing
five or six different things, so it's unclear to me what's not working.
Please let me know what you expect it to do that it isn't doing.
Thanks!

**Comment by Sphilippe7** [comment]
when I say it doesn't work, I mean it does not reset the board. The click handler works perfectly fine. It functions the same way as when I start a game but that is not exactly what I want it to do.
Upon load up, you have to:
-sign in
-then hit create game
-and after hitting create game, you can play.

The reset button I have, acts exactly like the create game. So essentially, you can hit the reset button after you sign in to start a new game. Which is not what I want.

I want it to function as a way to reset the board so after the winner is displayed, you hit the button and all the X's and O's disappear for you to now play another game.

The issue is not so much of my code not working, it is finding the right method to do what I want it to do. I've looked at other examples and tried other code but it still does not do what I want it to do. @raq929

**Comment by raq929** [comment]
Got it!
OK. So, first step is clear the HTML (the X's and O's) off the board, right? You look like you have some code that is supposed to do that, your $.each. So comment out everything in reset except that function. And comment out everything in it except `$(element).html('')`.
What happens? Does it do what you expect? Is there an error in the console?

**Comment by Sphilippe7** [comment]
The button is not doing anything. I opened up the console and got no response @raq929

**Comment by raq929** [comment]
If you put a console.log('hello') before the $.each, do you see it?

**Comment by raq929** [comment]
If not, you need to make sure the click handler is registered. Look at how you handled click handlers for sign in, or a different handler that's working.
If you can see the hello, then there's a problem in your each. Put a console log inside it, and see if you get anything.

**Comment by gaand** [comment]
This

``` js
  $('.resetgame').on('reset', resetgame);
```

and this

``` html
<button class="resetgame" type="button">Start a New Game!</button>
```

are a likely part of the problem.  The `'reset'` event only applies to forms.  You probably want a `'click'` event.

**Comment by raq929** [comment]
What's your status on this, Stanley?

**Comment by Sphilippe7** [comment]
So when I hit the reset button, I get an error and only one of the characters (either "X" or "O") disappears from a box

**Comment by raq929** [comment]
That's progress! Can you post the current code?

**Comment by Sphilippe7** [comment]
ok and now I have all the x's and o's to go away but there is still the problem of me not being able to click on the boxes after the x's and o's disappear

**Comment by Sphilippe7** [comment]
Ben told me I'm close but I've been trying and can't get it right

**Comment by Sphilippe7** [comment]
My logic file:

```js
const reset = function(){
gameBoardArray = [["box", "box", "box"],["box", "box", "box"],["box", "box", "box"]];
console.log('success');
    $('.box').off('click');
    $.each($('.box'), function(index, element) {
        $(element).html('');
        $(element).one('click', recordMove);
      } ) ;
      if (recordMove.turn === "x"){
         recordMove.turn = "o";
      } else {
        recordMove.turn = "x";
      }
    $('.ttt-box').show();
    $('.display-winner').hide('Player 2 Wins!');
};
```

**Comment by Sphilippe7** [comment]
This is the error. 
![screen shot 2016-11-21 at 5 53 50 pm](https://cloud.githubusercontent.com/assets/9563419/20504057/930b428c-b013-11e6-8244-0c7dbee8e81f.png)

**Comment by raq929** [comment]
From looking at the jquery documentation, you may want to be using `this` instead of `element` in your `.each(). `
http://api.jquery.com/each/
Your syntax should be more like:
```js
$('.box').each(function () {
  $(this).doSomething()
}
```
It looks like you confused it with this each: http://api.jquery.com/jQuery.each/
Which is totally understandable, as it's the first thing that comes up. They ought to have something in big red letters at the top of the page.

**Comment by Sphilippe7** [comment]
I changed all the "element" 's to "this" and still coming up with the same error

**Comment by raq929** [comment]
Can you post the new code? It should be in the format I showed you above. Also post `logic.js` line 133 if it's not that line.

**Comment by Sphilippe7** [comment]
```js
const reset = function(){
    $('.box').off('click');
    $.each($('.box'), function() {
        $(this).html('');
        $(this).one('click', recordMove);
      } ) ;
    $('.box').html('');
      if (recordMove.turn === "x"){
         recordMove.turn = "o";
      } else {
        recordMove.turn = "x";
      }
    $('.ttt-box').show();
    $('.display-winner').hide('Player 2 Wins!');
};
```

**Comment by raq929** [comment]
You need to use ` $('.box').each(function(){})` instead of $.each. They are different functions.

**Comment by Sphilippe7** [comment]
fixed the issue by adding gameboardarrary = [0][1] to the reset function. The reset function assumed that x and o was still in the boxes instead of empty ones. So I had to add all the empty boxes to the reset function. This is the visual picture of what I did with the help of @gaand .
![screen shot 2016-11-23 at 9 11 08 am](https://cloud.githubusercontent.com/assets/9563419/20564816/df6b2bd4-b15c-11e6-8c6b-2adebc4c81cb.png)
